### DISCUSSION TITLE
Queue::fake() not catch for ShouldBeUnique Jobs

### DISCUSSION THREAD

**Discussion by luca-rigutti** [laravel/framework#49953] [Q&A]
Hi,

I was tring to make a test of a Job dispatched (inside a workflow) and find that the ShouldBeUnique class that implement is not catch by `Queue::fake();`, but if removed (so the job is only `ShouldQueue`) it work.

This example is more light way for find the problem, because the Job is dispatched inside an artisan command, but it seem that is not the command the problem, but also if dispatched inside the test.

Anyone find something that can be setup to make it work?

My current versions are:

Laravel Version: 10.42.0
PHP Version: 8.3.2

This is the example

```php

<?php

namespace App\Jobs;

use Illuminate\Bus\Queueable;
use Illuminate\Contracts\Queue\ShouldQueue;
use Illuminate\Foundation\Bus\Dispatchable;
use Illuminate\Queue\InteractsWithQueue;
use Illuminate\Queue\SerializesModels;
use Illuminate\Contracts\Queue\ShouldBeUnique;

class Job implements ShouldQueue, ShouldBeUnique
{
    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;

    public function __construct(public int $id)
    {
        //
    }

    public function handle(): void
    {
        print("Test" . $this->id);
    }

    public function uniqueId(): string
    {
        return $this->id;
    }
}

```

```php

<?php

namespace Tests\Feature;

use Illuminate\Foundation\Testing\RefreshDatabase;
use Illuminate\Foundation\Testing\WithFaker;
use Tests\TestCase;

use Illuminate\Support\Facades\Queue;

use App\Jobs\Job;

class JobTest extends TestCase
{

    public function test_example_queue(): void
    {
        Queue::fake();
        
        Job::dispatch(1);

        Queue::assertPushed(Job::class, 1);
    }

}

```

**Comment by alexw23** [comment]
Noticed the same situation... this should be reported as a bug.

**Reply by alexw23** [reply]
Okay @luca-rigutti realised that this happens if `uniqueId` is missing.

I know you included this but just double check.
